internal/util/factory.go

```diff
@@ -35,6 +35,8 @@ type Factory interface {
 	// returned slice can contain IPv6, IPv4 or both, in no particular
 	// order.
 	PublicIPs(context.Context) ([]string, error)
+	// TF returns the target flags used to override the target
+	TF() target.TargetFlags
```

Comment:
Not sure about extending the factory. Maybe there is a better way

Reply:
Why dont't we use the same approach like kubectl does with https://pkg.go.dev/k8s.io/cli-runtime/pkg/genericclioptions.